### PR TITLE
Collect Wiki lifecycle events

### PR DESCRIPTION
**Pull request by m90**
Ticket https://phabricator.wikimedia.org/T345434

### CODE REVIEW

File: app/Jobs/UpdateWikiSiteStatsJob.php [hunk 1]
```diff
@@ -26,6 +29,31 @@ public function handle (): void
         }
     }
 
+    private function updateLifecycleEvents (Wiki $wiki): void {
+        $responses = Http::pool(fn (Pool $pool) => [
+            $pool->as('revisions')->withHeaders(['host' => $wiki->getAttribute('domain')])->get(
```
Comment (m90) [inline app/Jobs/UpdateWikiSiteStatsJob.php:34]: Theoretically, the value returned by this call will never change, so if we wanted to optimize for performance, we could also check if such a value has already been stored for the wiki and the only make the call if we need to.

I chose not to do this right now as we run this once a day only and performance is not an issue.

File: database/migrations/2023_09_19_094751_create_wiki_lifecycle_events_table.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('wiki_lifecycle_events', function (Blueprint $table) {
+            $table->id();
+            $table->timestamps();
+            $table->timestampTz('first_edited')->nullable();
```
Comment (m90) [inline database/migrations/2023_09_19_094751_create_wiki_lifecycle_events_table.php:19]: If I understand it correctly, all Wikis on wikibase.cloud run on UTC, so we could also drop the Timezone information here as it's always `+00:00`. Then again it's probably safer to already add it.

File: app/WikiLifecycleEvents.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+// This class is supposed to contain information about certain events
+// in a wiki's lifecycle, e.g. the time of the last edit. Sources for these
+// points in time can be chosen as needed.
+class WikiLifecycleEvents extends Model
```
Comment (m90) [inline app/WikiLifecycleEvents.php:11]: Naming suggestions are welcome. I think the current one is ok, but maybe there's something better out there still.

Comment (deer-wmde) [reply app/WikiLifecycleEvents.php:11]: I think that naming already fits quite good :)

**Review by deer-wmde — APPROVED** [review]
looks very good to me overall!

**Pull request merged**